conda/models/match_spec.py

```diff
@@ -240,8 +240,7 @@ def __eq__(self, other):
         return isinstance(other, MatchSpec) and self._eq_key() == other._eq_key()
 
     def __hash__(self):
-        # TODO: optional and target should not hash
-        return hash(self._specs_map)
+        return hash(self.name)
```

Comment:
@mcg1969 Just added this.  A lot of times when we're dealing with matchspecs, I've been using a dict where name is the key to dedupe/override a group of matchspecs.  Now we can just throw them all in a set, and update the set.  Are there problems with this that I'm missing?